### PR TITLE
(test) O3 2428: Test workflow for recording appointments

### PR DESCRIPTION
**Pull request by dilankavishka**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] My work includes tests or is validated by existing tests.

## Summary
Create an e2e test for create appointments for patient.

https://openmrs.atlassian.net/browse/O3-2428

## There is a error while Running the test

Screenshot 2023-12-02 at 12 22 42

### CODE REVIEW

**Comment by dilankavishka** [comment]
@jayasanka-sack

File: e2e/pages/index.ts [hunk 1]
```diff
@@ -2,3 +2,4 @@ export * from './allergies-page';
 export * from './vitals-and-biometrics-page';
 export * from './program-page';
 export * from './conditions-page';
+export * from './appointments-page';
```
[the remark this answers is not in the document]
Comment (RandilaP) [reply e2e/pages/index.ts:5]: Add a line below

File: e2e/specs/add-appointments.spec.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    });
+
+    await test.step('And I save the form', async () => {
+        await appointmentsPage.page.getByRole('button', { name: 'Save and close' }).click();
+    });
+
+    await test.step('Then I should see the notification', async () => {
+    });
+
+    await test.step('And I see the appointment in the list', async () => {
```
Comment (RandilaP) [inline e2e/specs/add-appointments.spec.ts:54]: same here

File: e2e/specs/add-appointments.spec.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    });
+
+    await test.step('And I click on the cancel appointment button', async () => {
+        await appointmentsPage.page.getByRole('button', { name: 'cancel appointment' }).click();
+    });
+
+    await test.step('Then I should see the notification', async () => {
```
Comment (RandilaP) [inline e2e/specs/add-appointments.spec.ts:61]: same here

File: e2e/specs/add-appointments.spec.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    });
+});
+
+test.afterEach(async ({ api }) => {
+    await deletePatient(api, patient.uuid);
+});
```
Comment (RandilaP) [inline e2e/specs/add-appointments.spec.ts:67]:
```suggestion
});

```

File: e2e/pages/appointments-page.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { Page } from "@playwright/test";
```
Comment (RandilaP) [inline e2e/pages/appointments-page.ts:1]:
```suggestion
import { type Page } from "@playwright/test";
```

File: e2e/specs/add-appointments.spec.ts [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    await test.step('Then I should see the notification', async () => {
+        await expect(appointmentsPage.page.getByText('saved')).toBeVisible();
+    });
+
+    await test.step('And I see the appointment in the list', async () => {
+    });
+        
+
+    await test.step('And I click on the edit appointment button', async () => {
+        await appointmentsPage.editButton().click();
```
Comment (RandilaP) [inline e2e/specs/add-appointments.spec.ts:41]: write a function for editButton() or write a locator for it

**Comment by RandilaP** [comment]
@dilankavishka Please don't use the description for posting errors. Could you post them in the conversation? After reviewing it I think it's causing it because there's a type error when trying to make an appointment page. I have made some suggestions. Please review them. If you need anything feel free to contact me through Slack or any other platform

**Comment by dilankavishka** [comment]
@jayasanka-sack @denniskigen @RandilaP 
I'm unable to complete this PR due to a save and close button issue.
https://openmrs.atlassian.net/browse/O3-2667

**Comment by denniskigen** [comment]
@dilankavishka, you should be unblocked by @chibongho's [work](https://github.com/openmrs/openmrs-esm-patient-chart/pull/1554).

**Comment by dilankavishka** [comment]
@denniskigen , yeah that's great! i'll complete my work just immediately.

**Comment by denniskigen** [comment]
Hey, @dilankavishka, do you have the bandwidth to move this work forward?

**Comment by dilankavishka** [comment]
@denniskigen Yeah i can do this. i'm almost 95% done on this issue. i will complete this just immediately. sorry for the late.

**Comment by jayasanka-sack** [comment]
Thanks for the update @dilankavishka. Could you please provide us an estimated completion date?

**Comment by dilankavishka** [comment]
@denniskigen @jayasanka-sack I promise to finish this before the end of the day. (before 26th of January)

**Comment by dilankavishka** [comment]
Apologies for the delay. I commit to completing this before January 31st. Thank you for your understanding.

**Comment by jayasanka-sack** [comment]
@dilankavishka could you clarify the current status or provide more details on the completion?

**Pull request closed without merging**